### PR TITLE
Re-usable AgeDropdown on ReportAbuseForm

### PR DESCRIPTION
**Pull request by Erin007**
There was a `AgeDropdown` component baked into the `ReportAbuseForm` as well as a nearly identical `AgeDropdown` component that we re-use in various places across the site.  I imported the re-usable `AgeDropdown` and deleted the version that was within the `ReportAbuseForm`.

### CODE REVIEW

**Review by epeach — COMMENTED** [review]
Yay for re-usability! I wonder if it makes sense to add a UI test here to ensure that a simple interaction with the age dropdown from the report abuse form works.

**Comment by Erin007** [comment]
@epeach good call! There is some [limited testing on the report abuse form](https://github.com/code-dot-org/code-dot-org/blob/8990f16a29d8a1cf9dba0ffb220bd424e8542810/apps/test/unit/code-studio/components/ReportAbuseFormTest.js), but no UI testing. Adding...

**Review by maddiedierker — APPROVED** [review]

File: dashboard/test/ui/features/report_abuse.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  Scenario: Reporting abuse as a signed-in teacher
+    Given I create a teacher named "Dumbledore"
+    Given I am on "http://studio.code.org/report_abuse"
+    # Email and age are already known and passed to the form
+    And I wait until element "#uitest-email" is not visible
+    And I wait until element "#uitest-age-selector" is not visible
+    And I select the "Other" option in dropdown "uitest-abuse-type"
+    And I type "Mudblood is an offensive term" into "#uitest-abuse-detail"
+    Then I click selector "#uitest-submit-report-abuse" once I see it
+    Then check that the URL matches "https://support.code.org/hc/en-us"
```
Comment (Erin007) [inline dashboard/test/ui/features/report_abuse.feature:31]: @epeach When you get a chance, can you take a peek at the UI tests I added for the report abuse form? I test that various account types can successfully submit;  but checking for the alerts when the required fields are left blank was starting to be a time suck. What do you think?

**Review by epeach — APPROVED** [review]
dumbledorecheers
Awesome tests!
I think the coverage is fine without checking for the alert, since the first scenario at least checks that it is possible to select an age and complete the form.

File: dashboard/app/controllers/report_abuse_controller.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class ReportAbuseController < ApplicationController
   AGE_CUSTOM_FIELD_ID = 24_024_923
 
   def report_abuse
-    unless Rails.env.development?
+    unless Rails.env.development? || Rails.env.test?
```
Comment (Erin007) [inline dashboard/app/controllers/report_abuse_controller.rb:23]: screen shot 2019-01-08 at 1 04 18 pm
We're blocked from sending Zendesk tickets, which reporting abuse does, from the test environment; so, I check the environment before submitting the ticket from the form.

**Pull request merged**